include/aspect/utilities.h

```diff
@@ -1060,6 +1066,41 @@ namespace aspect
      * entry in the list must match one of the allowed operations.
      */
     std::vector<Operator> create_model_operator_list(const std::vector<std::string> &operator_names);
+
+    /**
+    * A class that represents a point in a chosen coordinate system.
+    */
+    template <int dim>
+    class GeneralizedPoint
+    {
+      public:
+        GeneralizedPoint(Point<dim> position, const GeometryModel::Interface<dim> *geometry_model);
+
+        /**
+        * Returns the cartesian position of this point.
+        */
+        Point<dim> &get_position();
```

Comment:
Is this very useful to have? If you decide not to store the cartesian position and only use it to construct this instance, it would only contain the natural coordinates. Then we could call the class ``NaturalCoordinate`` or something and actually return it from the geometry models.

Reply:
hmm, that is a very interesting idea. I can not currently think of a situation where I would actually need the original position for my current use cases of this function. 

As a side note, we can always get the position back anyway by using the natural_to_cartesian_coordinates function. 

So the question is then do we want to 
1) remove the whole function and not store the position, 
2) leave the function but compute the position on the fly when needed, or 
3) leave the function as it is and store both the position and coordinate.

Reply:
I think it's ok to store both representations. (We probably won't store massive numbers of these objects anywhere, and even if we did, it would at worst triple the amount of memory that we use in `class Triangulation` to store vertex locations, for example.) So option 3.

Reply:
I did not suggest removing it because of memory concerns. The disadvantage is that if you want it to contain both representations, you always have to supply both of them (or cartesian + geometry model) in the constructor. I can see several situations where we want to use this in the future where it would be nice if I only need to supply the natural coordinates. Consider the case where I get some coordinates from the geometry, do some math with them, and then want to give them to a function that expects natural coordinates. How would I construct these with the current design?

Reply:
having one representation means you can compute the other in the constructor. I could add a constructor in which you give the natural coordinate, and it computes and stores the Cartesian coordinate internally for you.